### PR TITLE
Refactor(ui): tweak ui logic

### PR DESCRIPTION
**Pull request by YvonTre**
## Video State Changing Cycle

- All changable state(except for `currentTime`) must go through Vuex first, then to controller component and finally to base component. Here are some examples:

### When user wants to play/pause video

1. User clicks on the screen/press the space key.
2. Trigger global event `toggle-playback`.
3. `VideoCanvas`(As `BaseVideoPlayer`'s controller component) dispatch `play` or `pause` Vuex action.
4. Vuex state changes.
5. `VideoCanvas`'s Vuex `paused` getters changes.
6. `BaseVideoPlayer`'s `paused` props changed.
7. Video paused/played.

### When user wants to seek to some time

1. User clicks on the progress bar/press the left/right key.
2. Event listeners emit global seek event.
3. `VideoCanvas` change `BaseVideoPlayer`'s `currentTime` props.
4. `BaseVideoPlayer` change video's currentTime.
5. `BaseVideoPlayer` commit Vuex's `currentTime` mutations.

## Refactor Vuex

- Put mutation types and action types in `mutation-types.js` and `action-types.js`.
- Do not commit mutations directly, dispatch actions instead.
- Do validations in actions.
- Do not use state directluy components, use getters instead.

## Update `.eslintrc.js`

- Change complexity from 'warn' to 'error'.

## Clean up tests

- Delete all unnecessary `console.log`.
- Use sinon's fakeTimers when testing `requestAnimationFrame`.

#### `Error: [vuex] Do not mutate vuex store state outside mutation handlers.`

- Avoid mutating Vuex state directly.
- Avoid committing Vuex mutations. Use fake actions/mutations instead.

#### `[vuex] unknown getter: xxxxxx`

- Add lacked getters in some tests.

#### `[vuex] unknown action type: XXXXXX`

- Add lacked actions in some tests.

#### `[Vue warn]: Invalid prop: custom validator check failed for prop "defaultEvents".` && `[Vue warn]: Missing required prop: "defaultEvents"`

- Check propsData carefully when mounting/shallowMounting components.
- Avoid passing unnecessary props around components.

#### `[Vue warn]: Invalid default value for prop "xxxxxx": Props with type Object/Array must use a factory function to return the default value.`

- Use function like `() => {}/[];` as default props value for Object/Array.

#### `[Vue warn]: Avoid adding reactive properties to a Vue instance or its root $data at runtime - declare it upfront in the data option.`

- Avoid to use `wrapper.setData` on non-existed data properties.
- When clear unnecessary data properties, check whether unit tests have some legacy setData methods.

#### `TypeError: Cannot read property 'xxxxxx' of undefined/null`

- Improve function's robustness and better error handling capabilities.

## Rewrite TimeProgressBar

- Refactor a new TheProgressBar based on old TimeProgressBar.
- Use computed properties as much as possible.
- Use CSS to control style as much as possitble.
- Simplify hovered and played display logic.

## To-do

- Draft a unit-test style guide and FAQ.
- Refactor BaseSubtitle and SubtitleControl and their tests.
- Remove console.log completely.

### CODE REVIEW

File: src/renderer/components/PlayingView.vue [hunk 1]
```diff
@@ -16,14 +16,7 @@ export default {
...
   mounted() {
-    this.$bus.$emit('play');
-    this.$electron.ipcRenderer.send('callCurrentWindowMethod', 'setMinimumSize', [320, 180]);
-    this.$electron.ipcRenderer.send('callCurrentWindowMethod', 'setResizable', [true]);
-  },
-  computed: {
-    uri() {
-      return this.$store.getters.convertedSrcOfVideo;
-    },
+    this.$electron.remote.getCurrentWindow().setMinimumSize(320, 180);
```
Comment (ipy) [inline src/renderer/components/PlayingView.vue:19]: 这个调整原因是？

File: src/renderer/components/PlayingView/TimeProgressBar.vue [hunk 2]
```diff
@@ -255,13 +243,14 @@ export default {
...
     curProgressBarEdge() {
-      if (Number.isNaN(this.$store.state.PlaybackState.Duration)) {
+      if (Number.isNaN(this.duration)) {
         return 0;
       }
       const progressBarWidth = this.winWidth - FOOL_PROOFING_BAR_WIDTH;
-      return (this.$store.state.PlaybackState.CurrentTime
-        / this.$store.state.PlaybackState.Duration) * progressBarWidth;
+      return (this.currentTime
+        / this.duration) * progressBarWidth;
```
Comment (ipy) [inline src/renderer/components/PlayingView/TimeProgressBar.vue]: 类似这种应该可以合成一行了，之前换行应该是为了不超过单行字符数限制

File: src/renderer/store/mutation-types.js [hunk 3]
```diff
@@ -1,7 +1,34 @@
 export const Video = {
```
Comment (ipy) [inline src/renderer/store/mutation-types.js]: 除了这两个文件应该没有其他文件名使用连字符，建议统一格式

File: src/renderer/helpers/infoDB.js [hunk 4]
```diff
@@ -65,11 +65,10 @@ class InfoDB {
    * Replace a record if the given quickHash existed
    */
   static add(schema, data) {
-    console.log('adding');
     return idb.open('Info').then((db) => {
       const tx = db.transaction(schema, 'readwrite');
       tx.objectStore(schema).put(data);
-      return tx.complete.then(() => console.log('added'));
+      return tx.complete.then(() => 'added');
```
Comment (ipy) [inline src/renderer/helpers/infoDB.js]: 返回的这个字符串有用到么

File: src/renderer/components/PlayingView/NextVideo.vue [hunk 5]
```diff
@@ -15,7 +15,7 @@
       <div class="info">
         <div class="top">
           <div class="duration">{{ duration }}</div>
-          <div class="title">&nbsp· {{ title }}</div>
+          <div class="title">&nbsp;· Next in Playing List</div>
```
Comment (YuRiJinX) [inline src/renderer/components/PlayingView/NextVideo.vue]: 这个改回去啊

File: src/renderer/helpers/infoDB.js (hunk 4 above)
Comment (YuRiJinX) [reply src/renderer/helpers/infoDB.js]: 谭扬的pr已经改过了 @ipy

**Review by ipy — DISMISSED** [review]

**Review by ipy — APPROVED** [review]

**Pull request merged**